circus-train-avro/src/test/java/com/hotels/bdp/circustrain/avro/transformation/AbstractAvroSerDeTransformationTest.java

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright (C) 2016-2019 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.bdp.circustrain.avro.transformation;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.when;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import com.hotels.bdp.circustrain.api.event.EventReplicaTable;
+import com.hotels.bdp.circustrain.api.event.EventTableReplication;
+import com.hotels.bdp.circustrain.avro.conf.AvroSerDeConfig;
+
+@RunWith(MockitoJUnitRunner.class)
+public class AbstractAvroSerDeTransformationTest {
+
+  private class DummyAvroSerDeTransformation extends AbstractAvroSerDeTransformation {
+
+    protected DummyAvroSerDeTransformation(AvroSerDeConfig avroSerDeConfig) {
+      super(avroSerDeConfig);
+    }
+
+  }
+
+  private static final String EVENT_ID = "eventId";
+  private AbstractAvroSerDeTransformation transformation;
+
+  @Before
+  public void setUp() {
+    AvroSerDeConfig defaultConfig = new AvroSerDeConfig();
+    defaultConfig.setBaseUrl("default");
+    transformation = new DummyAvroSerDeTransformation(defaultConfig);
+  }
+
+  @Test
+  public void testOneReplicationsOverride() throws Exception {
+    EventTableReplication tableReplication = mockTableReplicaiton("overrideBaseUrl");
+    runLifeCycleSuccess(tableReplication);
+    assertThat(transformation.getAvroSchemaDestinationFolder(), is("overrideBaseUrl"));
+    assertThat(transformation.getEventId(), is(EVENT_ID));
+    assertThat(transformation.getTableLocation(), is("location"));
+  }
+
+  @Test
+  public void testMultipleReplicationsOverride() throws Exception {
+    EventTableReplication tableReplication = mockTableReplicaiton("overrideBaseUrl");
+    runLifeCycleSuccess(tableReplication);
+    assertThat("overrideBaseUrl", is(transformation.getAvroSchemaDestinationFolder()));
```

Comment:
any particular reason to reverse the order in these `assert` statements ?

Reply:
I think our convention is to have the thing being tested first and the constant comes after that.

Reply:
(hamcrest convention not ours) but yes I wrote them all wrong and switched I forgot to switch these.